From df9071487a8ff69370e932a1522054a69fa7cbb1 Mon Sep 17 00:00:00 2001
From: Alan Modra <amodra@gmail.com>
Date: Sun, 13 Feb 2022 09:56:51 +1030
Subject: [PATCH 08/91] PR28882, build failure with gcc-4.2 due to use of 0b
 literals

	PR 28882
	* elf/loongarch.h: Replace binary literals with hex.

(cherry picked from commit ccbaaa36174c2c3184344ee0e6db648491149a33)
---
 include/elf/loongarch.h | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/include/elf/loongarch.h b/include/elf/loongarch.h
index b835a3e98c7..1893ef5faae 100644
--- a/include/elf/loongarch.h
+++ b/include/elf/loongarch.h
@@ -93,14 +93,14 @@ RELOC_NUMBER (R_LARCH_GNU_VTENTRY, 58)
 END_RELOC_NUMBERS (R_LARCH_count)
 
 /* Processor specific flags for the ELF header e_flags field.  */
-#define EF_LOONGARCH_ABI_ILP32	    0b01
-#define EF_LOONGARCH_ABI_LP64	    0b11
-#define EF_LOONGARCH_ABI_MASK	    0b11
-
-#define EF_LOONGARCH_FLOAT_ABI_SOFT    0b001100
-#define EF_LOONGARCH_FLOAT_ABI_SINGLE  0b001000
-#define EF_LOONGARCH_FLOAT_ABI_DOUBLE  0b000000
-#define EF_LOONGARCH_FLOAT_ABI_MASK    0b111100
+#define EF_LOONGARCH_ABI_ILP32	    0x1
+#define EF_LOONGARCH_ABI_LP64	    0x3
+#define EF_LOONGARCH_ABI_MASK	    0x3
+
+#define EF_LOONGARCH_FLOAT_ABI_SOFT    0x0c
+#define EF_LOONGARCH_FLOAT_ABI_SINGLE  0x08
+#define EF_LOONGARCH_FLOAT_ABI_DOUBLE  0x00
+#define EF_LOONGARCH_FLOAT_ABI_MASK    0x3c
 
 #define EF_LOONGARCH_IS_LP64(abi) \
   ((abi & EF_LOONGARCH_ABI_MASK) == EF_LOONGARCH_ABI_LP64)
-- 
2.35.1

